pulpcore/pulpcore/app/viewsets/repository.py

```diff
@@ -125,34 +125,35 @@ class ImporterViewSet(NamedModelViewSet):
     queryset = Importer.objects.all()
     filter_class = ImporterFilter
 
-    def update(self, request, repository_name, name, partial=False):
+    def update(self, request, repository_pk, name, partial=False):
         importer = self.get_object()
         serializer = self.get_serializer(importer, data=request.data, partial=partial)
         serializer.is_valid(raise_exception=True)
         app_label = importer._meta.app_label
         async_result = tasks.importer.update.apply_async_with_reservation(
-            tags.RESOURCE_REPOSITORY_TYPE, repository_name,
-            args=(importer.id, app_label, serializer.__class__.__name__),
-            kwargs={'data': request.data, 'partial': partial}
+            tags.RESOURCE_REPOSITORY_TYPE, repository_pk,
+            kwargs={'importer_pk': importer.pk,
+                    'app_label': app_label,
+                    'serializer_name': serializer.__class__.__name__,
+                    'data': request.data,
+                    'partial': partial}
         )
         return OperationPostponedResponse([async_result], request)
 
-    def destroy(self, request, repository_name, name):
+    def destroy(self, request, repository_pk, name):
```

Comment:
same as above, name-> importer_pk
also for sync, publish update, publish delete